### ISSUE TITLE
Issue with Istio Getting Started Guide: NetworkPolicyHosts: Duplicate host entry

### ISSUE THREAD

**Issue by danwent** [cilium/cilium#4119]
This was encountered in the istio-proxy logs while running the Istio Getting Started Guide, and attempting to access the Istio ingress.  

Seems like the NetworkPolicyHosts config being passed to envoy is invalid, and has a duplicate entry: 

```
[2018-05-16 00:49:36.856][11][warning][config] bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:69] gRPC config for type.googleapis.com/cilium.NetworkPolicyHosts rejected: NetworkPolicyHosts: Duplicate host entry 100.96.7.36:13051
[2018-05-16 00:49:36.856][11][warning][config] cilium_host_map.cc:51] Bad NetworkPolicyHosts Configuration
[2018-05-16 00:49:36.858][11][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:188] gRPC config for type.googleapis.com/cilium.NetworkPolicyHosts update rejected: NetworkPolicyHosts: Duplicate host entry W.X.Y.Z:13051
```

Also has errors like this, which I think are just pointing to the same root cause: 

```
[2018-05-16 00:20:36.935][13][warning][config] cilium_network_policy.cc:84] Bad Network Policy Configuration
[2018-05-16 00:20:41.935][13][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:205] gRPC config stream closed: 1, 
[2018-05-16 00:20:41.935][13][warning][config] bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:66] gRPC update for type.googleapis.com/cilium.NetworkPolicy failed
```

**Comment by danwent** [comment]
Found some interesting mentions of the W.X.Y.Z IP in the cilium log ("not adding entry to ip cache; error unmarshaling data from key-value store" error="unexpected end of JSON input" ): 

```
cilium-hkb5j-2018-05-16T00:48:25.147333.log:level=debug msg="Emiting modify event for cilium/state/ip/v1/default/W.X.Y.Z=[123 34 73 80 34 58 34 49 48 48 46 57 54 46 55 46 51 54 34 44 34 73 68 34 58 49 51 48 53 49 44 34 77 101 116 97 100 97 116 97 34 58 34 99 105 108 105 117 109 45 103 108 111 98 97 108 58 100 101 102 97 117 108 116 58 105 112 45 49 57 56 45 49 56 45 53 45 55 50 46 101 99 50 46 105 110 116 101 114 110 97 108 58 51 57 49 48 55 34 125]" prefix=cilium/state/ip/v1 revision=2682874 watcher=endpointIPWatcher
cilium-hkb5j-2018-05-16T00:48:25.147333.log:level=debug msg="Emiting delete event for cilium/state/ip/v1/default/W.X.Y.Z=[]" prefix=cilium/state/ip/v1 revision=2684556 watcher=endpointIPWatcher
cilium-hkb5j-2018-05-16T00:48:25.147333.log:level=error msg="not adding entry to ip cache; error unmarshaling data from key-value store" error="unexpected end of JSON input" key=cilium/state/ip/v1/default/W.X.Y.Z value="[]"
cilium-thklj-2018-05-16T00:48:26.788657.log:level=debug msg="Emiting modify event for cilium/state/ip/v1/default/W.X.Y.Z=[123 34 73 80 34 58 34 49 48 48 46 57 54 46 55 46 51 54 34 44 34 73 68 34 58 49 51 48 53 49 44 34 77 101 116 97 100 97 116 97 34 58 34 99 105 108 105 117 109 45 103 108 111 98 97 108 58 100 101 102 97 117 108 116 58 105 112 45 49 57 56 45 49 56 45 53 45 55 50 46 101 99 50 46 105 110 116 101 114 110 97 108 58 51 57 49 48 55 34 125]" prefix=cilium/state/ip/v1 revision=2679682 watcher=endpointIPWatcher
cilium-thklj-2018-05-16T00:48:26.788657.log:level=debug msg="Emiting modify event for cilium/state/ip/v1/default/W.X.Y.Z=[123 34 73 80 34 58 34 49 48 48 46 57 54 46 55 46 51 54 34 44 34 73 68 34 58 49 51 48 53 49 44 34 77 101 116 97 100 97 116 97 34 58 34 99 105 108 105 117 109 45 103 108 111 98 97 108 58 100 101 102 97 117 108 116 58 105 112 45 49 57 56 45 49 56 45 53 45 55 50 46 101 99 50 46 105 110 116 101 114 110 97 108 58 51 57 49 48 55 34 125]" prefix=cilium/state/ip/v1 revision=2681288 watcher=endpointIPWatcher
cilium-thklj-2018-05-16T00:48:26.788657.log:level=debug msg="Emiting modify event for cilium/state/ip/v1/default/W.X.Y.Z=[123 34 73 80 34 58 34 49 48 48 46 57 54 46 55 46 51 54 34 44 34 73 68 34 58 49 51 48 53 49 44 34 77 101 116 97 100 97 116 97 34 58 34 99 105 108 105 117 109 45 103 108 111 98 97 108 58 100 101 102 97 117 108 116 58 105 112 45 49 57 56 45 49 56 45 53 45 55 50 46 101 99 50 46 105 110 116 101 114 110 97 108 58 51 57 49 48 55 34 125]" prefix=cilium/state/ip/v1 revision=2682874 watcher=endpointIPWatcher
cilium-thklj-2018-05-16T00:48:26.788657.log:level=debug msg="Emiting delete event for cilium/state/ip/v1/default/W.X.Y.Z=[]" prefix=cilium/state/ip/v1 revision=2684556 watcher=endpointIPWatcher
cilium-thklj-2018-05-16T00:48:26.788657.log:level=error msg="not adding entry to ip cache; error unmarshaling data from key-value store" error="unexpected end of JSON input" key=cilium/state/ip/v1/default/W.X.Y.Z value="[]"
```

**Comment by danwent** [comment]
CC @k-amy

**Comment by danwent** [comment]
I'm also seeing errors of the type: 

```
cilium-wjzl5-2018-05-16T00:48:27.720235.log:level=debug msg="error deleting service before adding it" error="Service frontend not found X.64.222.113:9080" serviceName="X.64.222.113:9080"
```

**Comment by rlenglet** [comment]
This log is the key: `NetworkPolicyHosts update rejected: NetworkPolicyHosts: Duplicate host entry W.X.Y.Z:13051`.
This is a duplicate of https://github.com/cilium/cilium/issues/3825, which was fixed in https://github.com/cilium/cilium/pull/3865.
That fix is in Cilium v1.0.0 and beyond.
